### ISSUE TITLE
Compile crashes on ndarray.all().all()

### ISSUE THREAD

**Issue by Suor** [serge-sans-paille/pythran#475]
Here is my python file:
```python
#pythran export all_true(bool[][])

def all_true(data):
    return data.all().all()
```
And here is what I get:
```
CRITICAL Leave the vessel! Women and children first!
C++ compiler failed to compile translated code.
E: Compile error!

******** Command line was: ********
'c++' '/tmp/tmp5A7NGB.cpp' '-I/usr/include/python2.7' '-I/home/suor/projects/.virtualenvs/new-stargeo/lib/python2.7/site-packages/numpy/core/include/numpy' '-I/home/suor/projects/.virtualenvs/new-stargeo/lib/python2.7/site-packages/.' '-I/home/suor/projects/.virtualenvs/new-stargeo/lib/python2.7/site-packages/pythran' '-DENABLE_PYTHON_MODULE' '-DUSE_GMP' '-DNDEBUG' '-O2' '-g' '-fno-math-errno' '-std=c++11' '-O2' '-shared' '-o' 'analysis_pythran.so' '-L/usr/lib/python2.7/config-x86_64-linux-gnu' '-lpthread' '-ldl' '-lutil' '-lpython2.7' '-lboost_python' '-lgmp' '-lgmpxx' '-fPIC'

******** Output :  ********

In file included from /tmp/tmp5A7NGB.cpp:12:0:
/home/suor/projects/.virtualenvs/new-stargeo/lib/python2.7/site-packages/pythran/pythonic/numpy/all.hpp: In instantiation of ‘bool pythonic::numpy::all(const E&, pythonic::types::none_type) [with E = bool]’:
/home/suor/projects/.virtualenvs/new-stargeo/lib/python2.7/site-packages/pythran/pythonic/numpy/all.hpp:68:9:   required from ‘decltype (pythonic::numpy::all((forward<Types>)(pythonic::numpy::proxy::all::operator()::types)...)) pythonic::numpy::proxy::all::operator()(Types&& ...) const [with Types = {bool}; decltype (pythonic::numpy::all((forward<Types>)(pythonic::numpy::proxy::all::operator()::types)...)) = bool]’
/tmp/tmp5A7NGB.cpp:53:77:   required from ‘typename __pythran_analysis_pythran::all_true::type<argument_type0>::result_type __pythran_analysis_pythran::all_true::operator()(const argument_type0&) const [with argument_type0 = pythonic::types::ndarray<bool, 2ul>; typename __pythran_analysis_pythran::all_true::type<argument_type0>::result_type = bool]’
/tmp/tmp5A7NGB.cpp:59:51:   required from here
/home/suor/projects/.virtualenvs/new-stargeo/lib/python2.7/site-packages/pythran/pythonic/numpy/all.hpp:30:104: error: request for member ‘begin’ in ‘expr’, which is of non-class type ‘const bool’
                 return _all(expr.begin(), expr.end(), utils::int_<types::numpy_expr_to_ndarray<E>::N>());
                                                                                                        ^
/home/suor/projects/.virtualenvs/new-stargeo/lib/python2.7/site-packages/pythran/pythonic/numpy/all.hpp:30:104: error: request for member ‘end’ in ‘expr’, which is of non-class type ‘const bool’
/home/suor/projects/.virtualenvs/new-stargeo/lib/python2.7/site-packages/pythran/pythonic/numpy/all.hpp:30:104: error: no type named ‘type’ in ‘struct std::enable_if<false, void>’
```

**Comment by serge-sans-paille** [comment]
I can reproduce the issue with master branch, seems related to a missing specialisation of ``numpy.all``

**Comment by serge-sans-paille** [comment]
Confirmed :-) ``numpy.all`` was not defined for a scalar input. I'll check similar functions before committing the fix!

**Comment by serge-sans-paille** [comment]
Oooops^{TM} Fixed in 7ffbbc0f5d9d7359efc2a4e7cca61896598834ea but I forgot to tell you.
